### PR TITLE
add optional message

### PR DESCRIPTION
**Pull request by maks-us**
Hi 
I thought you may like to add this PR to your project. 
It is adding optional message to each entry. 

examples: 
```check in 18:30 "working on ticket 7865"``` <- adding message 
```check in "working on ticket 7865"``` <- adding message 
```check in 18:30 ```
```check in```

check out "message" is optional. It will rewrite check in "message"
```check out 20:00 "ticket 888"```
```check out "ticket 888"```
```check out 20:00```
```check out ```

example:
```
➜  checkin.py git:(master) ./check in 17:00 "working on ticket #673"
November 2017
====================
Sun, 12. Nov	17:00 -  NOW  = 1:42:00  'working on ticket #673'

                         Total: 1h 42m


Total missing: 158h 18m

➜  checkin.py git:(master) ./check out "finished ticket #673"
November 2017
====================
Sun, 12. Nov	17:00 - 18:43 = 1:42:26  'finished ticket #673'

                         Total: 1h 42m


Total missing: 158h 17m
```

p.s. I think about simple integration with alfred app on mac

### CODE REVIEW

File: check [hunk 1]
```diff
@@ -28,18 +28,30 @@ def get_time(t):
...
-def checkin(when=None):
+def checkin(when_or_message=None, message=None):
     if db:
         last_checkin = db[-1]
         assert 'out' in last_checkin, "Already checked in"
-    db.append({'in': get_time(when)})
+    try:
+        get_time(when_or_message)
+    except:
+        message = when_or_message
```
Comment (jonashaag) [inline check:38]: Wouldn't it be cleaner to do an `is None` check here instead?

**Review by jonashaag — COMMENTED** [review]
Didn't expect any contribution to this repository, thanks :)

File: check (hunk 1 above)
Comment (maks-us) [reply check:38]: I really like this small tool :) 👍 
Are you talking about replacing ```try / except``` with if condition ?

Comment (jonashaag) [reply check:38]: yes

**Comment by jonashaag** [comment]
OK, this looks much worse now. I didn't understand that your initial idea was to be able to use both "checkin.py 12:45" and "checking.py 'Fixed shit'" syntaxes. This makes a lot of sense and your initial implementation is much better than this. I thought you'd only ever pass time or both time and message, but you actually want to pass time, message, or time and message. My bad. I'll merge it with the old implementation if you remove the latest commit :)

**Comment by maks-us** [comment]
I agree. 
It become slightly over-engineered :) 
I returned first version. 👍

**Comment by jonashaag** [comment]
Thanks! Would you mind adding the new feature to the README? An example with time only, message only, and both time and message would be great. Also, explain what happens if you provide a message both on checkin and checkout.

**Comment by maks-us** [comment]
Hi @jonashaag  
I keep thinking about mobile interface for this app. I really like simplicity. 
What do you think about web version? Could it be usefully for you?

It would have the same idea - one field and check in button or one field and check out button. The field text format is the same like cli. So it would be possible to use cli via api and phone and web. Web smartphone friendly UI + ability to download hours.json file for backup/analytic. 

My current use case is tracking time and keeping myself focus. I see myself using it on phone via browser. I really like the simplicity. What is your main use case ?

Let me know your thoughts about it.

**Comment by jonashaag** [comment]
That could be fun but I think you should do it in another project. This project is basically a quick hack with terrible architecture. Things get much more complicated with API and multiple clients.

I wonder if there aren't dead simple time tracking tools out there already.

**Comment by maks-us** [comment]
I'll take a look it. So far i saw only complex system. Definitely it would be a separate GitHub project.

**Pull request merged**